In [ ]:
# Imports

import pandas as pd
from pandas import json_normalize
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
pd.set_option('display.max_columns', None)

In [ ]:
# Load NC Teams Dataset

nc_df = pd.read_csv('dataset_nc_teams_2025.csv')

In [20]:
#Create Scatter Plot

color_map = {
    'Duke': '#003087',
    'North Carolina': '#7BAFD4',
    'NC State': '#CC0000',
    'Wake Forest': '#9E7E38'}

# Scatter (colored by team)
fig = px.scatter(
    nc_df,
    x='points',
    y='firstDowns',
    color='team',
    color_discrete_map=color_map,
    title='Points vs First Downs (NC Teams - Game Level)',
    labels={
        'points': 'Points Scored',
        'firstDowns': 'First Downs'
    }, template='plotly_dark')

# Single overall trendline
trendline = px.scatter(
    nc_df,
    x='points',
    y='firstDowns',
    trendline='ols')

# Add trendline
fig.add_trace(trendline.data[1])

# Style the trendline
fig.data[-1].line.color = 'white'
fig.data[-1].name = 'Trendline'

# Layout
fig.update_layout(
    title_x=0.5,
    legend_title_text='Team'
)

fig.show()

In [21]:
# Pearson's R (Correlation)
corr = nc_df['points'].corr(nc_df['firstDowns'])
print(f"Pearson's r: {corr:.3f}")

Pearson's r: 0.609


A moderate positive correlation (r = 0.61) was observed between first downs and points scored, suggesting that teams that sustain drives more effectively tend to generate higher scoring outputs.